hops/model.py

```diff
@@ -106,6 +106,55 @@ def get_model(name, version, project_name=None):
 
     raise ModelNotFound("No model with name: {} and version {} could be found".format(name, version))
 
+
+def download_model(name, version=None, project_name=None, overwrite=False):
+    """
+    Download from the Hopsworks Models dataset an archive (zip file) containning the model artifcacts.
+    You first need to use the project.connect function to connect to Hopsworks.
+    If the Models dataset ths model resides in is a shared dataset from another project,
+    then you need to specify the name of the project that owns the Models dataset was shared from.
+
+    For example if you run this:
+
+    >>> from hops import model
+    >>> model.download_model('mnist')
+
+    Args:
+        :name: name of the model
+        :version: version of the model. If ommitted, all versions of the model will be included in the archive.
+        :project_name name of the project parent of the model. By default, this project is the current project running
+        the experiment
+        :overwrite: Whether to overwrite the model archive  file if it already exists
+
+    Returns:
+        A zip file containing the model artficats
+
+    Raises:
+        :ModelArchiveExists: if the model archive that contains the model artifcats already exists
+    """
+    if project_name is None:
+        project_name = hdfs.project_name()
+
+    # Check if model archive already exists and if it should be deleted, otherwise return an error
+    model_dir = '/Projects/' + project_name + "/Models/" + name
+    if version is not None:
+        model_dir += "/" + str(version)
+        name += str(version)
+    archive_path = model_dir + ".zip"
+    name += ".zip"
```

Comment:
With the upcoming transformers, the artifacts are stored within the version folder (e.g /Models/mnist/1/1.zip). I think it's a better option to keep /Models/mnist folder clean. But for now it doesn't matter, I can adjust this in the upcoming PR.

Reply:
The zip API doesn't let us overwrite the output path. I could move the file afterwards but it's too much.

Reply:
I agree but I would like this PR to be strictly about the client library so it goes in the codebase fast andalos  it's easy for users to upgrade the hops python library version, otherwise they have to redeploy Hopsworks. We should do as you suggest in another JIRA, can you open it?

Reply:
Alright then